resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -1197,6 +1206,34 @@ final DnsServerAddressStream newNameServerAddressStream(String hostname) {
         return dnsServerAddressStreamProvider.nameServerAddressStream(hostname);
     }
 
+    private static boolean isSpecialInvalid(final String hostname, final Promise<?> promise) {
+        if (INVALID.equalsIgnoreCase(hostname)) {
```

Comment:
Actually, RFC says:

> The domain "invalid." and any names falling within ".invalid." are special in the ways listed below.

https://tools.ietf.org/html/rfc6761#section-6.4

I think subdomains are not that important. We can start with support for `invalid.` only, because more detailed checks will introduce unnecessary overhead for each resolution. 

Do we need support for `invalid` without the last dot?

Reply:
I think what you did is good enough .

Reply:
After thinking about it more, I don't think it's worth placing something on the hot path just for testing purposes. Users can mock the resolver or filter it to have the same behavior in tests. "should" statement from the RFC does not obligate us for something.
Closing the PR?

Reply:
sure why not...